app/models/katello/event_queue.rb

```diff
@@ -0,0 +1,25 @@
+module Katello
+  class EventQueue < Katello::Model
+    self.table_name = 'katello_event_queue'
+
+    @repository_types = {}
+
+    def self.pop_event
+      first = Katello::EventQueue.order(:created_at => 'asc').limit(1)
+      events = Katello::EventQueue.where(:object_id => first.pluck(:object_id), :event_type => first.pluck(:event_type)).destroy_all
+      events.first
+    end
+
+    def self.push_event(event_type, id)
+      self.create!(:event_type => event_type, :object_id => id)
```

Comment:
apparently my brain checked out when i was creating this class :)  That should be event types